### ISSUE TITLE
[Feature] Cassandra storage provider support

### ISSUE THREAD

**Issue by jacek99** [jobrunr/jobrunr#6]
**Is your feature request related to a problem? Please describe.**

Not possible to use JobRunr on apps that use Cassandra, without introducing another storage layer.

**Describe the solution you'd like**

Allow Cassandra as a background job storage provider

**Describe alternatives you've considered**

Adding PgSQL to an existing Cassandra app. Possible, but cumbersome.

**Additional context**

You would probably need to use lightweight transactions in Cassandra.

**Comment by rdehuyss** [comment]
Hi @jacek99 ,

I'll be looking into adding support for Cassandra and keep you posted using this issue.

**Comment by rdehuyss** [comment]
Hi @jacek99 ,

I added support for Cassandra: 
-  the source code is here: [https://github.com/rdehuyss/jobrunr-storage-cassandra](https://github.com/rdehuyss/jobrunr-storage-cassandra)
- the jar and pom file are here: [https://github.com/rdehuyss/jobrunr-storage-cassandra/releases/tag/v0.9](https://github.com/rdehuyss/jobrunr-storage-cassandra/releases/tag/v0.9)
- the StorageProvider passes the StorageProviderTest, which I see as a requirement when adding support for a new StorageProvider.

I did not add it in the main repository of JobRunr as I feel that I'm currently misusing Cassandra as a queue. Since the state of a Job changes a lot and this is one of the main query criteria, updates in Cassandra are also not possible. I'm using Batch statements to do a delete which is then followed by an update combined in a lightweight transaction.

That is why, for now, I added support in an external repository. If I see that demand for Cassandra grows, I can revisit this decision. But, I want to keep the code of JobRunr clean and mean and I don't have that feeling yet for the Cassandra support - probably also because of my lack of knowledge around Cassandra.

I'm looking forward to your feedback.
Ronald
